vineyardctl failed to be built on MacOS on Arm

Describe your problem
---------------------

https://cirrus-ci.com/task/5215989055356928?logs=build_vineyardctl#L38

```
make -C k8s vineyardctl
go: creating new go.mod: module tmp
go: downloading sigs.k8s.io/controller-tools v0.11.0
go: downloading github.com/spf13/cobra v1.6.1
go: downloading github.com/gobuffalo/flect v0.3.0
go: downloading k8s.io/apiextensions-apiserver v0.26.0
go: downloading k8s.io/apimachinery v0.26.0
go: downloading golang.org/x/tools v0.4.0
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading github.com/fatih/color v1.13.0
go: downloading k8s.io/api v0.26.0
go: downloading gopkg.in/yaml.v3 v3.0.1
go: downloading sigs.k8s.io/yaml v1.3.0
go: downloading github.com/spf13/pflag v1.0.5
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading k8s.io/utils v0.0.0-20221107191617-1a15be271d1d
go: downloading github.com/google/gofuzz v1.1.0
go: downloading k8s.io/klog/v2 v2.80.1
go: downloading sigs.k8s.io/structured-merge-diff/v4 v4.2.3
go: downloading github.com/mattn/go-colorable v0.1.9
go: downloading github.com/mattn/go-isatty v0.0.14
go: downloading sigs.k8s.io/json v0.0.0-20220713155537-f223a00ba0e2
go: downloading golang.org/x/sys v0.3.0
go: downloading gopkg.in/inf.v0 v0.9.1
go: downloading github.com/go-logr/logr v1.2.3
go: downloading github.com/json-iterator/go v1.1.12
go: downloading golang.org/x/mod v0.7.0
go: downloading golang.org/x/net v0.4.0
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/modern-go/reflect2 v1.0.2
go: downloading golang.org/x/text v0.5.0
/Users/admin/go/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x102888a30]
goroutine 13 [running]:
go/types.(*Checker).handleBailout(0x14000e6b200, 0x14000f71d18)
	/opt/homebrew/Cellar/go/1.22.0/libexec/src/go/types/check.go:367 +0x9c
panic({0x102b61360?, 0x1030fbe90?})
	/opt/homebrew/Cellar/go/1.22.0/libexec/src/runtime/panic.go:770 +0x124
go/types.(*StdSizes).Sizeof(0x0, {0x102c2bb70, 0x103104620})
	/opt/homebrew/Cellar/go/1.22.0/libexec/src/go/types/sizes.go:228 +0x320
go/types.(*Config).sizeof(...)
	/opt/homebrew/Cellar/go/1.22.0/libexec/src/go/types/sizes.go:333
go/types.representableConst.func1({0x102c2bb70?, 0x103104620?})
	/opt/homebrew/Cellar/go/1.22.0/libexec/src/go/types/const.go:76 +0x9c
go/types.representableConst({0x102c31db0, 0x1030d07c8}, 0x14000e6b200, 0x103104620, 0x14000f70b58)
	/opt/homebrew/Cellar/go/1.22.0/libexec/src/go/types/const.go:92 +0x138
go/types.(*Checker).representation(0x14000e6b200, 0x14000e757c0, 0x103104620)
	/opt/homebrew/Cellar/go/1.22.0/libexec/src/go/types/const.go:256 +0x68
go/types.(*Checker).implicitTypeAndValue(0x14000e6b200, 0x14000e757c0, {0x102c2bb70, 0x103104620})
	/opt/homebrew/Cellar/go/1.22.0/libexec/src/go/types/expr.go:375 +0x340
go/types.(*Checker).assignment(0x14000e6b200, 0x14000e757c0, {0x102c2bb70, 0x103104620}, {0x10299d4f0, 0xe})
	/opt/homebrew/Cellar/go/1.22.0/libexec/src/go/types/assignments.go:52 +0x23c
go/types.(*Checker).exprInternal(0x14000e6b200, 0x0, 0x14000e757c0, {0x102c2fbe0, 0x140008c7e00}, {0x0, 0x0})
	/opt/homebrew/Cellar/go/1.22.0/libexec/src/go/types/expr.go:1175 +0x1e58
go/types.(*Checker).rawExpr(0x14000e6b200, 0x0, 0x14000e757c0, {0x102c2fbe0?, 0x140008c7e00?}, {0x0?, 0x0?}, 0x0)
	/opt/homebrew/Cellar/go/1.22.0/libexec/src/go/types/expr.go:979 +0x12c
go/types.(*Checker).expr(0x14000e6b200, 0x0?, 0x14000e757c0, {0x102c2fbe0?, 0x140008c7e00?})
	/opt/homebrew/Cellar/go/1.22.0/libexec/src/go/types/expr.go:1513 +0x38
go/types.(*Checker).varDecl(0x14000e6b200, 0x14000391c80, {0x14000400960, 0x1, 0x1}, {0x0, 0x0}, {0x102c2fbe0, 0x140008c7e00})
	/opt/homebrew/Cellar/go/1.22.0/libexec/src/go/types/decl.go:521 +0x140
go/types.(*Checker).objDecl(0x14000e6b200, {0x102c36f78, 0x14000391c80}, 0x0)
	/opt/homebrew/Cellar/go/1.22.0/libexec/src/go/types/decl.go:194 +0x7ec
go/types.(*Checker).packageObjects(0x14000e6b200)
	/opt/homebrew/Cellar/go/1.22.0/libexec/src/go/types/resolver.go:693 +0x468
go/types.(*Checker).checkFiles(0x14000e6b200, {0x14000e47100, 0x2, 0x2})
	/opt/homebrew/Cellar/go/1.22.0/libexec/src/go/types/check.go:408 +0x164
go/types.(*Checker).Files(...)
	/opt/homebrew/Cellar/go/1.22.0/libexec/src/go/types/check.go:372
sigs.k8s.io/controller-tools/pkg/loader.(*loader).typeCheck(0x14000356fc0, 0x1400093a460)
	/Users/admin/go/pkg/mod/sigs.k8s.io/controller-tools@v0.11.0/pkg/loader/loader.go:286 +0x2d8
sigs.k8s.io/controller-tools/pkg/loader.(*Package).NeedTypesInfo(0x1400093a460)
	/Users/admin/go/pkg/mod/sigs.k8s.io/controller-tools@v0.11.0/pkg/loader/loader.go:99 +0x44
sigs.k8s.io/controller-tools/pkg/loader.(*TypeChecker).check(0x14000d4dc20, 0x1400093a460)
	/Users/admin/go/pkg/mod/sigs.k8s.io/controller-tools@v0.11.0/pkg/loader/refs.go:268 +0x304
sigs.k8s.io/controller-tools/pkg/loader.(*TypeChecker).check.func1(0x0?)
	/Users/admin/go/pkg/mod/sigs.k8s.io/controller-tools@v0.11.0/pkg/loader/refs.go:262 +0x58
created by sigs.k8s.io/controller-tools/pkg/loader.(*TypeChecker).check in goroutine 1
	/Users/admin/go/pkg/mod/sigs.k8s.io/controller-tools@v0.11.0/pkg/loader/refs.go:260 +0x230
make: *** [generate] Error 2
```

Additional context
------------------

Add any other context about the problem here.

--- a/.cirrus.yml
+++ b/.cirrus.yml
@@ -14,7 +14,8 @@ macosx_arm64_wheel_task:
 
   install_deps_script:
     - brew update
-    - brew install coreutils golang pyenv python
+    # go 1.22 doesn't work, see also: https://github.com/golang/go/issues/37602
+    - brew install coreutils go@1.21 pyenv python
 
   install_various_python_script:
     - pyenv install 3.8.18
@@ -154,7 +155,9 @@ macosx_arm64_wheel_task:
     sudo rm -rf /tmp/install
 
   build_vineyardctl_script:
-    - make -C k8s vineyardctl
+    - |
+      export PATH=$(brew --prefix go@1.21)/bin:$PATH
+      make -C k8s vineyardctl
 
   vineyardctl_artifacts:
     path: k8s/vineyardctl
@@ -197,7 +200,7 @@ macosx_arm64_wheel_task:
       for py in 3.8.18 3.9.18 3.10.13 3.11.8 3.12.2; do
         current_python=$(pyenv root)/versions/$py/bin/python
         echo "Python is: $current_python, $($current_python --version)"
-        $current_python -m pip install delocate wheel
+        $current_python -m pip install delocate setuptools wheel
         rm -rf build/lib* build/bdist.* python/vineyard/*.dylib python/vineyard/*.so
         mkdir -p build
         cd build

